File: app/src/main/java/com/ytempest/wanandroid/http/bean/BaseResp.java
package com.ytempest.wanandroid.http.bean;

/**
 * @author heqidu
 * @since 2020/7/4
 */
public class BaseResp<T> {
    // 0：成功，1：失败
    private int errorCode;

    private String errorMsg;

    private T data;

    public int getErrorCode() {
        return errorCode;
    }

    public void setErrorCode(int errorCode) {
        this.errorCode = errorCode;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    /*Ext*/

    public static final int SUCCESS = 0;
    public static final int FAIL = 1;

    /**
     * 判断请求是否成功
     */
    public boolean isSuccess() {
        return errorCode == SUCCESS;
    }
}

File: app/src/main/java/com/ytempest/wanandroid/http/bean/BaseResp.kt
package com.ytempest.wanandroid.http.bean

/**
 * @author heqidu
 * @since 21-2-8
 */
class BaseResp<T> {

    // 0：成功，1：失败
    private var errorCode = 0

    private var errorMsg: String? = null

    private var data: T? = null

    fun getErrorCode(): Int {
        return errorCode
    }

    fun setErrorCode(errorCode: Int) {
        this.errorCode = errorCode
    }

    fun getErrorMsg(): String? {
        return errorMsg
    }

    fun setErrorMsg(errorMsg: String?) {
        this.errorMsg = errorMsg
    }

    fun getData(): T? {
        return data
    }

    fun setData(data: T) {
        this.data = data
    }

    /*Ext*/
    val SUCCESS = 0
    val FAIL = 1

    /**
     * 判断请求是否成功
     */
    fun isSuccess(): Boolean {
        return errorCode == SUCCESS
    }
}